Сетка страниц и крупных блоков на флексах

eatokareva opened the pull request:
Сетка страниц и крупных блоков на флексах

---
:mortar_board: [Сетка страниц и крупных блоков на флексах](https://up.htmlacademy.ru/htmlcss/33/user/2011429/tasks/7)

:boom: https://htmlacademy-htmlcss.github.io/2011429-device-33/5/

eatokareva committed:
flex layout

Флекс-верстка крупных блоков

eatokareva committed:
flex layout catalog

Флекс верстка крупных блоков внутренней страницы

eatokareva committed:
made sticky footer

eatokareva committed:
corrections flex layout

eatokareva committed:
correction header-middle

eatokareva committed:
correct micro flex

Откорректированы микросетки на флексах

eatokareva committed:
correct flex

откорректировано расположение флекс сеток и учтено переполнение

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -50,24 +105,159 @@ body {
+  align-items: baseline;
+  margin: 0;
+  padding: 0;
+  width: 240px;
```
А зачем это ограничение?

eatokareva replied:
я решила, что раз у нас рядом с кнопкой "войти" должна быть свободная зона, чтобы поместилась ссылка на Пользователя, то правый блок меню надо ограничить

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -50,24 +105,159 @@ body {
+  display: none;
+}
+
+.menu-user-list {
```
Тут еще нужно добавить возможность переполнения контента.

eatokareva replied:
да, я уже добавила, но добавила, когда грид делала, поэтому не выгрузила еще

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -50,24 +105,159 @@ body {
 
 .menu-link {
+  display: block;
+  padding: 15px 0;
```
А зачем эти отступы?

eatokareva replied:
в макете блок Каталог товаров отражен как блок с отступами в 15px , я сделала ссылки блочными элементами с padding, чтобы при нажатии было удобнее пользователю.

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -77,88 +267,316 @@ header .container {
+  margin-right: 15px;
+}
+
+.button-dropdown {
```
Не забываем на все кнопки ставить `cursor: pointer`

eatokareva replied:
ок, точно, поправлю

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -77,88 +267,316 @@ header .container {
+.service-list {
+  margin: 0;
+  padding: 0;
+  padding-bottom: 64px;
```
А этот отступ зачем?

eatokareva replied:
есть padding у всего блока service, но и у блока с перечнем услуг тоже есть нижний отступ, хотя возможно правильнее было сделать padding у блока с описанием?

vlad-yermolayev replied:
Да.

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -169,64 +587,244 @@ header .container {
   color: #ffffff;
 }
 
+.footer-menu-list {
```
Тут нужны отступы между пунктами списка, на случай если их будет больше. А также нет переноса пунктов при переполнении.

eatokareva replied:
Переполнение да, поставила уже, а вот по поводу отступов: я их делала через space-between, так неправильно, надо через margin?

vlad-yermolayev replied:
Если будет много пунктов, то они прижмутся в плотную. Нужен отступ внешний.

eatokareva replied:
ок

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -241,25 +839,116 @@ footer {
+  padding-right: 330px;
+  display: flex;
+  justify-content: space-between;
+  width: 150px;
```
Тут лучше использовать `min-width`, контента может быть больше и он будет вылетать за пределы своего родителя при свойстве `width`

eatokareva replied:
ок, поняла

vlad-yermolayev commented on styles/styles.css:
```diff
@@ -169,64 +587,244 @@ header .container {
+.button-contacts {
+  box-sizing: border-box;
+  display: block;
+  width: 260px;
```
Тут лучше использовать min-width, контента может быть больше и он будет вылетать за пределы своего родителя при свойстве width

eatokareva replied:
ок

eatokareva replied:
Наверно тогда и на другие кнопки тоже надо поставить min-width?

vlad-yermolayev replied:
Да

eatokareva replied:
Начала делать, но вопрос насчет меню в хедере: у нас есть кнопки Найти, Войти/Выйти, если у них поставить min-width то при длинном наименовании поедет все меню, нельзя у них поставить не min-width, а min-height, чтобы  в случае чего кнопка стала выше, но не шире, т.к. в этом случае меню не поедет?

eatokareva committed:
correct footer-menu inner-page

eatokareva committed:
correct footer-menu index-page

eatokareva committed:
correct buttons, padding and margin, add flex-wrap

Откорректированы кнопки (min-width/height, добавлен cursor: pointer), добавлены отступы в меню в футере, в блоке service, добавлена возможность переполнения флексов  wrap

eatokareva committed:
add modal form delivery

Добавлено модальное окно с формой доставки под заказ

eatokareva committed:
create grid catalog-card

 Карточки товаров отображены через grid

eatokareva committed:
correct login-logout
